modules/internal/String.chpl

```diff
@@ -209,7 +209,7 @@ module String {
   use ChapelStandard;
   use CTypes;
   use ByteBufferHelpers;
-  use BytesStringCommon;
+  public use BytesStringCommon;
```

Comment:
What requires this to be publicly used now?  (and do we want that?  Are the things not intended for end-users appropriately `private`?)